packages/flame/lib/src/components/navigator.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:collection/collection.dart';
+import 'package:flame/src/components/component.dart';
+import 'package:flame/src/components/page.dart';
+
+class Navigator extends Component {
+  Navigator({
+    required Map<String, Page> pages,
+    required this.initialPage,
+  }) : _pages = pages;
+
+  final String initialPage;
+  final Map<String, Page> _pages;
+  final List<Page> _currentPages = [];
+
+  void showPage(String name) {
+    final page = _pages[name];
+    assert(page != null, 'Page "$name" is not known to the Navigator');
+    final activePage = _currentPages.lastOrNull;
+    if (page! == activePage) {
+      return;
+    }
+    if (!page.isMounted) {
+      add(page);
+    }
+    _currentPages.remove(page);
+    _currentPages.add(page);
+    _fixPageOrder();
+    activePage?.deactivate();
+    page.activate();
+  }
+
+  void popPage() {
```

Comment:
```suggestion
  void pop() {
```
If we want to follow the same style as the flutter navigator

Reply:
I kinda find the `pushPage() / popPage()` pair more symmetric and balanced than `pushPage() / pop()`.

I don't think it would be confusing to the users: even if they try to write `pop()`, the autocomplete will tell them that it should be `popPage()`.

Reply:
Now renamed into `pushRoute()` / `popRoute()`.

Reply:
Now renamed into `pushNamed` / `pop` :)